micro_dl/inference/image_inference.py

```diff
@@ -72,33 +90,35 @@ def __init__(self,
         :param int gpu_id: GPU number to use. -1 for debugging (no GPU)
         :param float/None gpu_mem_frac: Memory fractions to use corresponding
             to gpu_ids
+        :param TorchPredictor torch_predictor: predictor object which handles
+            transporting data to a PyTorch model for inference.
         """
-        # Use model_dir from inference config if present, otherwise use train
-        if 'model_dir' in inference_config:
-            model_dir = inference_config['model_dir']
-        else:
-            model_dir = train_config['trainer']['model_dir']
-
-        if 'save_folder_name' in inference_config:
-            self.save_folder_name = inference_config['save_folder_name']
-        else:
-            self.save_folder_name = 'predictions'
-
-        if 'model_fname' in inference_config:
-            model_fname = inference_config['model_fname']
+        # create model and load weights
+        self.torch_predictor = torch_predictor
+        
+        # Use model_dir from torch config.
+        if 'model_dir' in self.torch_predictor.network_config:
+            model_dir = self.torch_predictor.network_config['model_dir'] 
         else:
-            # If model filename not listed, grab latest one
-            fnames = [f for f in os.listdir(inference_config['model_dir'])
-                      if f.endswith('.hdf5')]
-            assert len(fnames) > 0, 'No weight files found in model dir'
-            fnames = natsort.natsorted(fnames)
-            model_fname = fnames[-1]
-
+            raise AssertionError('model_dir must be specified in torch config for inference')
+        
         self.config = train_config
         self.model_dir = model_dir
         self.image_dir = inference_config['image_dir']
 
+        if 'save_folder_name' in inference_config:
```

Comment:
I agree, we should do more careful handling of paths. That being said if someone forgets the '/' on essentially any path provided in the config files the code will break, as paths are treated as absolute in a number of other places (I ran into this the first time I ran the code as a user). 

I am relying on users to know that specifying a save folder starting with '/' will make it global, and not doing so will make it local (in this case I've treated local as local to the model dir, rather than the running dir, as I saw Soorya had expected this behavior, so it seemed intuitive to me). Is there another robust way you were thinking of that a user can specify that they want an abs vs. local path used as the save directory?